crates/ruff_workspace/src/options.rs

```diff
@@ -482,6 +482,17 @@ pub struct Options {
     /// field (e.g., `requires-python = ">=3.8"`). If Ruff is configured via
     /// `ruff.toml` or `.ruff.toml`, no such inference will be performed.
     pub target_version: Option<PythonVersion>,
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            # Enable preview features
+            preview = true
+        "#
+    )]
+    /// Whether to enable preview mode. When preview mode is enabled, Ruff will
+    /// use unstable rules and fixes.
+    pub preview: Option<bool>,
```

Comment:
So.. I think we could make this `Option<PreviewMode>` too but I'm not entirely sure how to make that play nicely with the `JsonSchema`. I'm also not sure if we want `preview = enabled | disabled` or `preview = true | false` for users.

Reply:
I quickly looked into this but don't see a good solution for it other than implementing `Serialize`, `Deserialize` and `JsonSchema` manually on `PreviewMode` if we want `preview = true | false` (preview = enabled | disabled) should work out of the box.

Reply:
Thanks! I'm happy to leave it as-is then